src/ClinicalDecisionRules/Interface/RuleLibrary/RuleManager.php

```diff
@@ -494,62 +508,46 @@ function deleteRuleFilter($rule, $guid)
-            sqlStatement(
-                "INSERT INTO clinical_rules (id, pid, active_alert_flag, passive_alert_flag, cqm_flag, amc_flag, patient_reminder_flag, developer, funding_source, release_version, web_reference, bibliographic_citation, linked_referential_cds ) " .
-                    "VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?) ",
-                array(
-                        $ruleId,
-                        0,
-                        in_array(RuleType::ActiveAlert, $types) ? 1 : 0,
-                        in_array(RuleType::PassiveAlert, $types) ? 1 : 0,
-                        in_array(RuleType::CQM, $types) ? 1 : 0,
-                        in_array(RuleType::AMC, $types) ? 1 : 0,
-                        in_array(RuleType::PatientReminder, $types) ? 1 : 0,
-                        $developer,
-                        $funding,
-                        $release,
-                        $web_ref,
-                        $bibliographic_citation,
-                        $linked_referential_cds
-                    )
-            );
-
-            // do label
-            $this->doRuleLabel(false, "clinical_rules", $ruleId, $title);
-            return $ruleId;
+        $exists = !empty($rule->id);
+        if (!$exists) {
+            $dataToPersist = $rule->getSummaryDataToPersist();
+            // seems odd to have this here.
+            $dataToPersist['id'] = $this->getNextRuleId();
+            $dataToPersist['pid'] = 0;
+            $values = str_repeat("?,", count($dataToPersist) - 1) . "?";
+            $sql = "INSERT INTO `clinical_rules` (" . implode(",", array_keys($dataToPersist))
+                    . ") VALUES (" .  $values . ")";
+            $bind = array_values($dataToPersist);
+            QueryUtils::sqlInsert($sql, $bind);
+            $ruleId = $dataToPersist['id'];
         } else {
-            // edit
-            // update flags
-            sqlStatement(self::SQL_UPDATE_FLAGS, array(
-                in_array(RuleType::ActiveAlert, $types) ? 1 : 0,
-                in_array(RuleType::PassiveAlert, $types) ? 1 : 0,
-                in_array(RuleType::CQM, $types) ? 1 : 0,
-                in_array(RuleType::AMC, $types) ? 1 : 0,
-                in_array(RuleType::PatientReminder, $types) ? 1 : 0,
-                $developer,
-                $funding,
-                $release,
-                $web_ref,
-                $bibliographic_citation,
-                $linked_referential_cds,
-                $rule->id ));
-
-            // update title
-            sqlStatement(self::SQL_UPDATE_TITLE, array( $title,
-                $ruleId ));
-            return $ruleId;
+            $dataToPersist = $rule->getSummaryDataToPersist();
+            $ruleId = $dataToPersist['id'];
+            unset($dataToPersist['id']);
+            $sql = "UPDATE `clinical_rules` SET ";
+            foreach ($dataToPersist as $key => $value) {
+                $sql .= $key . " = ?,";
```

Comment:
just making sure escaping not needed on $dataToPersist keys

Reply:
Nope, column names are hard-coded and do not come from user data so we are safe here.